### PR TITLE
BAVL-256: Use new Book A Video Link API

### PR DESCRIPTION
**Pull request by smcveigh941**
Adds a feature toggle for Book a Video Link functionality.

When the toggle is `true`:
- Video link bookings are created via the Book A Video Link API rather than Whereabouts API.
- Locations offered for the appointment are video link enabled locations only
- "Other" is removed as an option in the list of courts
- An additional field is added to collect the court hearing type (reference data from BAVL API)

### CODE REVIEW

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: server/controllers/appointmentController.ts [hunk 1]
```diff
@@ -145,7 +159,7 @@ export default class AppointmentController {
         const appointmentsToCreate: AppointmentDefaults = {
           bookingId,
           appointmentType,
-          locationId: Number(location),
+          locationId: appointmentType === 'VLB' && config.featureToggles.bookAVideoLinkEnabled ? vlbLocation : location,
           startTime,
```
Comment (timharrison-moj) [inline server/controllers/appointmentController.ts:163]: Does this this need to be as it was before - with `Number(location)`

File: server/controllers/appointmentController.ts [hunk 2]
```diff
@@ -296,17 +310,27 @@ export default class AppointmentController {
         clientToken,
         user.activeCaseLoadId,
       )
-      courts.push({ id: 'other', name: 'Other' })
+
+      if (!config.featureToggles.bookAVideoLinkEnabled) {
+        ;(courts as CourtLocation[]).push({ id: 'other', name: 'Other' })
+      }
```
Comment (timharrison-moj) [inline server/controllers/appointmentController.ts:316]: semi-colon doesn't look right here?

File: server/data/interfaces/whereaboutsApi/Appointment.ts [hunk 3]
```diff
@@ -33,7 +33,7 @@ export interface AppointmentForm {
 export interface AppointmentDefaults {
   bookingId: number
   appointmentType: string
-  locationId: number
+  locationId: string
   startTime: string
```
Comment (timharrison-moj) [inline server/data/interfaces/whereaboutsApi/Appointment.ts:37]: Related to the previous question - altered from number to string.

**Review by timharrison-moj — APPROVED** [review]
LGTM

File: server/data/interfaces/whereaboutsApi/Appointment.ts [hunk 4]
```diff
@@ -49,11 +49,12 @@ export interface PrePostAppointmentDetails {
...
   formValues?: {
     preAppointment?: string
-    preAppointmentLocation?: number
+    preAppointmentLocation?: string
     postAppointment?: string
-    postAppointmentLocation?: number
+    postAppointmentLocation?: string
     court?: string
     otherCourt?: string
+    hearingType?: string
```
Comment (timharrison-moj) [inline server/data/interfaces/whereaboutsApi/Appointment.ts:57]: Question really - when BVLS switch is true, we should never let the user provide the `otherCourt` value - they must always select from the list. Do that hold up?

**Review by timharrison-moj — DISMISSED** [review]
LGTM

**Review by davidatmoj — APPROVED** [review]

**Pull request merged**